### PR TITLE
Modified HeapPage.java and HeapFile.java

### PR DESCRIPTION
**Pull request by Rithesh17**
The previous files had a few minor mistakes, and 2 test cases had failed. I have corrected them.

### CODE REVIEW

**Review by Priiyam — APPROVED** [review]

**Review by Ram-Aditya — APPROVED** [review]
Good To Go.

**Comment by sbshah97** [comment]
Why has this Pull Request not been merged?

**Review by manishsbhoopalam — APPROVED** [review]

**Comment by Ram-Aditya** [comment]
@salman-bhai We thought we needed to wait for your approval as we had requested a review from your side.
Sorry about that!We `ll go ahead with the merge now then.

**Pull request merged**
